lib/customer.ts

```diff
@@ -0,0 +1,222 @@
+   * @constructor
+   */
+  constructor(json:string) {
+    let tmp = JSON.parse(json);
+    if ("email" in tmp) this.email = tmp.email;
+    else throw new Error("Missing parameter: email");
+
+    if ("lastname" in tmp) this.lastname = tmp.lastname;
+    else throw new Error("Missing parameter: lastname");
+
+    if ("firstname" in tmp) this.firstname = tmp.firstname;
+    else throw new Error("Missing parameter: firstname");
+
+    this.stripeCusid = null;
+    this.sources = [];
+    if ("stripeCusid" in tmp) this.stripeCusid = tmp.stripeCusid;
+    if ("sources" in tmp) this.sources = tmp.sources;
+
+    this.map={};
+    this.map[Payment.card]='card';
+    this.map[Payment.sepa]='sepa_debit';
+    this.map[Payment.bitcoin]='bitcoin';
+  }
+
+  /**
+  * ## customer.create()
+  * Async constructor of customer
+  * @param {string} email
+  * @param {string} lastname
+  * @param {string} firstname
+  * @returns {Boolean} Check result
+  */
+  static create(email:string, lastname:string, firstname:string) {
+    return stripe.customers.create({
+      description: lastname+' '+firstname,
+      email: email
+    }).then((customerStripe) => {
+      var custJson = JSON.stringify({
+        email:email,
+        lastname:lastname,
+        firstname:firstname,
+        stripeCusid:customerStripe.id
+      });
+      return new Customer(custJson);
+    }).catch(parseError);
+  }
+
+  /**
+  * ## customer.save()
+  * Serialize the object into JSON
+  * @returns {string} Customer object in JSON
+  */
+  save() {
+    var json:string;
+
+    return JSON.stringify(this);
+  }
+
+  /**
+  * ## customer.addPayment()
+  * Add method of payment for the customer
+  * @param {Source} sourceData Source object containing all the informatio
+  * needed for payment creation
+  * @param {string} token Some source require token to be created
+  * @returns {any} Promise for the source creation
+  */
+  addPayment(sourceData:Source, token?:string) {
+    if (!(sourceData.type in this.map))
+      throw new Error("Unknown payment type");
+
+    var newSourceData = {...sourceData}; // copy sourceData
+    if (newSourceData.type == Payment.card) {
+      return stripe.customers.createSource(this.stripeCusid,{ source: token }).then((card) => {
+        var newCard:Card = {
+          type:Payment.card,
+          sourceId:card.id,
+          owner:newSourceData.owner,
+          brand:card.brand,
+          exp_year:card.exp_year,
+          exp_month:card.exp_month,
+          last4:card.last4
+        }
+        this.sources.push(newCard);
+      }).catch(parseError);
+    } else {
+      throw new Error("Unknown payment type");
+    }
+  }
+
+  /**
+  * ## customer.updatePayment()
+  * Update a payment's method of the customer
+  * @param {string} sourceId Stripe id of the source
+  * @param {Source} sourceData New data for the source
+  */
+  updatePayment(sourceId:string, sourceData:Source) {
+    let index:number=-1;
+    for (let i in this.sources) {
```

Comment:
argg, mieux d'utiliser `this.sources.findIndex(elem => elem.sourceId===sourceId)`